libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -290,25 +291,40 @@ void CompilerUtils::abiDecode(TypePointers const& _typeParameters, bool _fromMem
 
 					// Check that the data pointer is valid and that length times
 					// item size is still inside the range.
-					Whiskers templ(R"({
-						if gt(ptr, 0x100000000) { <revertStringPointer> }
-						ptr := add(ptr, base_offset)
-						let array_data_start := add(ptr, 0x20)
-						if gt(array_data_start, input_end) { <revertStringStart> }
-						let array_length := mload(ptr)
-						if or(
+					vector<string> localvars{"input_end", "base_offset", "offset", "ptr", "dst"};
```

Comment:
Since we argue above that old code generation is fine, do we actually need to change this? Or is the idea not to have to call the revert functions here? I didn't verify the code yet pending clarifying that.

Reply:
This change is needed because the revert strings utility function does not return the code anymore but only the name of a yul function to be called.